tools/project_manager/projectmgr.php

```diff
@@ -87,61 +76,50 @@
 elseif($show_view == "user_all")
 {
     $condition = "username = '$pguser'";
-    $_SESSION['search_data'] = array('project_manager' => $pguser);
+    // adjust $_GET so will work corectly with refine search and sort and navigate
+    // keep "user_all" or we won't know it is user all
+    $_GET = array_merge($_GET, array('project_manager' => $pguser));
     // TRANSLATORS: Abbreviation for Project Manager
     $sub_title = sprintf(_("All PM projects: %s"), $pguser);
 }
 elseif ($show_view == "user_active")
 {
     $condition = "$PROJECT_IS_ACTIVE_sql AND username = '$pguser'";
-    set_session_user_active();
+    $_GET = array_merge($_GET, array('project_manager' => $pguser, 'state' => array_diff($PROJECT_STATES_IN_ORDER, array(PROJ_SUBMIT_PG_POSTED, PROJ_DELETE))));
     // TRANSLATORS: Abbreviation for Project Manager
     $sub_title = sprintf(_("Active PM projects: %s"), $pguser);
 }
-else // $show_view == 'p_search' or 'search'
+else // $show_view == 'search'
 {
     $search_form = new ProjectSearchForm();
-    $condition = $search_form->get_condition($show_view);
-    // change search to p_search to use saved data if needed later
-    // since we are not re-using GET data
-    $show_view = 'p_search';
+    $condition = $search_form->get_condition();
     $sub_title = _("Search Results");
 }
 
-$search_results = new ProjectSearchResults($show_view, "PM");
+$search_results = new ProjectSearchResults("PM");
 
 echo_manager_links();
 echo "<h1>", _("Project Management"), "</h1>\n";
 // possibly show message, but don't exit
 check_user_can_load_projects(false);
 show_news_for_page('PM');
-echo "\n<h2 id='head'>$sub_title</h2>\n";
-echo_shortcut_links($show_view, $search_results);
+echo "\n<h2>$sub_title</h2>\n";
+echo_shortcut_links($show_view);
 
 if($show_view == "blank")
     exit();
 
 $search_results->render($condition);
 
-function set_session_user_active()
-{
-    global $pguser, $PROJECT_STATES_IN_ORDER;
-
-    $_SESSION['search_data'] = array(
-        'project_manager' => $pguser,
-        'state' => array_diff($PROJECT_STATES_IN_ORDER, array(PROJ_SUBMIT_PG_POSTED, PROJ_DELETE)),
-    );
-}
-
-function create_shortcut_link($text, $url, $show_view="")
+function create_shortcut_link($text, $show_val, $show_view="")
 {
-    if($show_view != $url)
-        return "<a href='{$_SERVER['PHP_SELF']}?show=$url'>$text</a>";
+    if($show_view != $show_val)
+        return "<a href='{$_SERVER['PHP_SELF']}?show=$show_val'>$text</a>";
```

Comment:
Again, we should try to remove `$_SERVER['PHP_SELF']` here. I don't think it's needed.